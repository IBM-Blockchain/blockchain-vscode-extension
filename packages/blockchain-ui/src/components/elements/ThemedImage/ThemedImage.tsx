import React from 'react';

interface IProps {
    altText: string;
    darkImg: any;
    lightImg: any;
    id: string;
    className?: string;
}

function ThemedImage(props: IProps): JSX.Element {
    const style: string = (props.className) ? props.className : '';
    return (
        <div id={`${props.id}-themed-image`}>
            <img src={props.lightImg} alt={props.altText} className={style}/>
            <img src={props.darkImg} alt={props.altText} className={style}/>
        </div>
    );
}

export default ThemedImage;
