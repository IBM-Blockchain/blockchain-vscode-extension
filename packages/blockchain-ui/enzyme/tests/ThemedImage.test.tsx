import React from 'react';
import renderer from 'react-test-renderer';
import { mount, ReactWrapper } from 'enzyme';
import chai from 'chai';
import sinonChai from 'sinon-chai';

import ThemedImage from '../../src/components/elements/ThemedImage/ThemedImage';
import lightImg from '../../src/resources/devLearnerLight.svg';
import darkImg from '../../src/resources/devLearnerDark.svg';

chai.should();
chai.use(sinonChai);

interface IProps {
    altText: string;
    darkImg: any;
    lightImg: any;
    id: string;
    className?: string;
}

describe('ThemedImage component', () => {
    it('should render the expected snapshot', () => {
        const component: any = renderer
            .create(<ThemedImage lightImg={lightImg} darkImg={darkImg} altText='' id='test-themed-image'/>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });

    it('should add any additional styles provided', () => {
        const component: ReactWrapper<IProps> = mount(<ThemedImage lightImg={lightImg} darkImg={darkImg} altText='' id='test-themed-image' className='some-class'/>);
        component.find('img').at(0).hasClass('some-class').should.equal(true);
        component.find('img').at(1).hasClass('some-class').should.equal(true);
    });
});
