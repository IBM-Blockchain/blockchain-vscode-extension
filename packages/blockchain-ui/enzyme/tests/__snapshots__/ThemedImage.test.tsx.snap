// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedImage component should render the expected snapshot 1`] = `
<div
  id="test-themed-image-themed-image"
>
  <img
    alt=""
    className=""
    src="test-file-stub"
  />
  <img
    alt=""
    className=""
    src="test-file-stub"
  />
</div>
`;
